packages/s3-credentials-endpoint/main.tf

```diff
@@ -0,0 +1,161 @@
+data "aws_caller_identity" "current" {}
+
+resource "aws_dynamodb_table" "access_tokens" {
+  name         = "${var.prefix}-s3-credentials-access-tokens"
+  billing_mode = "PAY_PER_REQUEST"
+  hash_key     = "accessToken"
+
+  attribute {
+    name = "accessToken"
+    type = "S"
+  }
+}
+
+data "aws_iam_policy_document" "assume_lambda_role" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "s3_credentials_lambda" {
+  name                 = "${var.prefix}-S3CredentialsLambda"
+  assume_role_policy   = data.aws_iam_policy_document.assume_lambda_role.json
+  permissions_boundary = var.permissions_boundary
+}
+
+data "aws_iam_policy_document" "s3_credentials_lambda" {
+  statement {
+    actions   = ["lambda:InvokeFunction"]
+    resources = [var.sts_credentials_lambda_arn]
+  }
+
+  statement {
+    actions   = ["dynamodb:GetItem"]
+    resources = [aws_dynamodb_table.access_tokens.arn]
+  }
+
+  statement {
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+    resources = ["arn:aws:logs:*:*:*"]
+  }
+  statement {
+    actions = [
+      "ec2:CreateNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DeleteNetworkInterface"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_role_policy" "s3_credentials_lambda" {
+  policy = data.aws_iam_policy_document.s3_credentials_lambda.json
+  role   = aws_iam_role.s3_credentials_lambda.id
+}
+
+resource "aws_lambda_function" "s3_credentials" {
+  function_name = "${var.prefix}-s3-credentials-endpoint"
+  # TODO Fetch this from ... somewhere. Or package it with the module zip file? Probably the better option
+  filename         = "${path.module}/dist/src.zip"
+  source_code_hash = filebase64sha256("${path.module}/dist/src.zip")
+  handler          = "index.handler"
+  role             = aws_iam_role.s3_credentials_lambda.arn
+  runtime          = "nodejs8.10"
+  timeout          = 10
+  memory_size      = 320
+  vpc_config {
+    subnet_ids = var.subnet_ids
+    security_group_ids = var.ngap_sgs
+  }
+  environment {
+    variables = {
+      DISTRIBUTION_REDIRECT_ENDPOINT = "https://${var.rest_api.id}.execute-api.${var.region}.amazonaws.com/${var.stage_name}/${var.redirect_path}"
+      # TODO Remove this stub value
+      # public_buckets            = ""
```

Comment:
This should probably be set.

Reply:
https://github.com/nasa/cumulus/pull/1086/commits/1af739dea0a39b44c85d33dd9768f3e93f698d15